eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -142,15 +143,23 @@ object Transactions {
   }
 
   /** Fee for an un-trimmed HTLC. */
-  def htlcOutputFee(feeratePerKw: Long): Satoshi = weight2fee(feeratePerKw, htlcOutputWeight)
+  def htlcOutputFee(feeratePerKw: Long): MilliSatoshi = weight2feeMsat(feeratePerKw, htlcOutputWeight)
 
-  def commitTxFee(dustLimit: Satoshi, spec: CommitmentSpec): Satoshi = {
+  /**
+   * While fees are generally computed in Satoshis (since this is the smallest on-chain unit), it may be useful in some
+   * cases to calculate it in MilliSatoshi to avoid rounding issues.
+   * If you are adding multiple fees together for example, you should always add them in MilliSatoshi and then round
+   * down to Satoshi.
+   */
+  def commitTxFeeMsat(dustLimit: Satoshi, spec: CommitmentSpec): MilliSatoshi = {
```

Comment:
Since we have a nice typing mechanism which removes all ambiguity, I would have changed the signature of `commitTxFee` to return `MilliSatoshi`, and let callers truncate to satoshis when they need it.

Same comment for `weight2fee`.

Reply:
I did that at first, but it was impacting too much code that really only wanted to deal with sats (all the parts related to actual on-chain transactions related to funding/closing)...